conan/tools/cmake/toolchain/toolchain.py

```diff
@@ -184,6 +184,18 @@ def content(self):
         content = relativize_generated_file(content, self._conanfile, "${CMAKE_CURRENT_LIST_DIR}")
         return content
 
+    def _find_cmake_exe(self):
+        for req in self._conanfile.dependencies.build.values():
```

Comment:
Better ``direct_build`` only?
```suggestion
        for req in self._conanfile.dependencies.direct_build.values():
```